examples/basic_infection/README.md

```diff
@@ -25,6 +22,20 @@ Infected individuals schedule their recovery at time `t + infected period`. The
 flowchart LR
 S(Susceptible) --FoI--> I(Infected) --inf. period--> R(Recovered)
 ```
+## Simulation architecture
+The simulation executaion first initializes the model parameters and prepares the simulation modules.
+
+### Initialization and modules
+
+ - *parameters module*: this module manages data for parameter values for population size, force of infection, and infection duration. At initialization, all parameter variables are set to a specific value.
+ - *person module*: contains a unique ID for each person that identifies each person in the simulation.
+ - *person_infection_status module*: this module connects each person ID with a specific value for a person's infection status, which could be one of Susceptible, Infected, or Recovered.
+ - *population manager module*: The population manager module is in charge of setting up the population. At initialization, the module reads the population size (N) parameter and creates N persons with unique ids (1..N) and sets their initial infection status to Susceptible.
+ - *transmission module*: This module is in charge of spreading the infection through the population. At initialization, it schedules an infection attempt for time 0. An infection attempt consists of choosing at random a susceptible individual from the population and setting their infection status to **infected**.
```

Comment:
Infection attempt events also schedule follow-on infection attempt events with exponential inter-arrival times (assuming you mean a Poisson point process here), so there is another parameter at play here too.